pkg/runtime/field_export_reconciler.go

```diff
@@ -0,0 +1,673 @@
+// Copyright Amazon.com Inc. or its affiliates. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License"). You may
+// not use this file except in compliance with the License. A copy of the
+// License is located at
+//
+//     http://aws.amazon.com/apache2.0/
+//
+// or in the "license" file accompanying this file. This file is distributed
+// on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+// express or implied. See the License for the specific language governing
+// permissions and limitations under the License.
+
+package runtime
+
+import (
+	"context"
+	"fmt"
+	"strings"
+
+	"github.com/go-logr/logr"
+	jq "github.com/itchyny/gojq"
+	"github.com/pkg/errors"
+	corev1 "k8s.io/api/core/v1"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	runtime "k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/types"
+	ctrlrt "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	k8sctrlutil "sigs.k8s.io/controller-runtime/pkg/controller/controllerutil"
+	"sigs.k8s.io/controller-runtime/pkg/predicate"
+
+	ackv1alpha1 "github.com/aws-controllers-k8s/runtime/apis/core/v1alpha1"
+	ackcondition "github.com/aws-controllers-k8s/runtime/pkg/condition"
+	ackcfg "github.com/aws-controllers-k8s/runtime/pkg/config"
+	ackerr "github.com/aws-controllers-k8s/runtime/pkg/errors"
+	ackmetrics "github.com/aws-controllers-k8s/runtime/pkg/metrics"
+	"github.com/aws-controllers-k8s/runtime/pkg/requeue"
+	ackrtcache "github.com/aws-controllers-k8s/runtime/pkg/runtime/cache"
+	ackrtlog "github.com/aws-controllers-k8s/runtime/pkg/runtime/log"
+	acktypes "github.com/aws-controllers-k8s/runtime/pkg/types"
+)
+
+const (
+	fieldExportFinalizerString = "finalizers.services.k8s.aws/FieldExport"
+)
+
+var (
+	pathDoesNotExistError = errors.New("path does not exist in this object")
+)
+
+// Remove the global to make it easier to mock
+var (
+	UnstructuredConverter runtime.UnstructuredConverter = runtime.DefaultUnstructuredConverter
+)
+
+// fieldExportReconciler is responsible for reconciling the state of any field
+// export CRs that target resources supported by the current controller.
+// It implements the upstream controller-runtime `Reconciler` interface.
+type fieldExportReconciler struct {
+	reconciler
+	// rd is only used if binding to an ACK resource (not `FieldExport`)
+	rd *acktypes.AWSResourceDescriptor
```

Comment:
Yeah I wanted to ask you about the best practices for something like this. I wanted to use this as some sort of flag that the reconciler is acting on a particular AWS resource (not a `FieldExport` type). I was using the fact that a pointer can be nil-able as this flag. Do you think it would be better to just explicitly put a boolean, instead?